### PR TITLE
python3Packages.seasonal: init at 0.3.1

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/seasonal/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+buildPythonPackage rec {
+  pname = "seasonal";
+  version = "0.3.1";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "welch";
+    repo = "seasonal";
+    # There are no tags or releases, but this commit corresponds to the 0.3.1 version
+    # PyPI project contains only a wheel
```
Comment (mbalatsko) [inline pkgs/development/python-modules/seasonal/default.nix:23]: I truly believe that this is a better and safer way to go, as we are able to run tests

**Comment by mbalatsko** [comment]
Result of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` on x86_64-linux:

```
2 packages built:
python310Packages.seasonal python311Packages.seasonal
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.seasonal python311Packages.seasonal

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

File: pkgs/development/python-modules/seasonal/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    setuptools
+    pytest-runner
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+    scipy
+    pandas
+    matplotlib
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/seasonal/default.nix:40]:
```suggestion
  ];
  
  passthru.optional-dependencies = {
    csv = [
      pandas
    ];
    plot = [
      matplotlib
    ];
  };
```
https://github.com/welch/seasonal/blob/2a2396014d46283d0c7aff34cde5dafb6c462c58/setup.py#L38-L44

File: pkgs/development/python-modules/seasonal/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    repo = "seasonal";
+    # There are no tags or releases, but this commit corresponds to the 0.3.1 version
+    # PyPI project contains only a wheel
+    rev = "2a2396014d46283d0c7aff34cde5dafb6c462c58";
+    hash = "sha256-8YedGylH70pI0OyefiS1PG1yc+sg+tchlgcuNvxcNqE=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    pytest-runner
```
Comment (natsukium) [inline pkgs/development/python-modules/seasonal/default.nix:32]:
```suggestion
```
Can we patch this line to remove pytest-runner?
https://github.com/welch/seasonal/blob/2a2396014d46283d0c7aff34cde5dafb6c462c58/setup.py#L48

File: pkgs/development/python-modules/seasonal/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+, pandas
+, matplotlib
+, pytestCheckHook
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "seasonal";
+  version = "0.3.1";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/seasonal/default.nix:17]:
```suggestion
  pyproject = true;
```

Comment (mbalatsko) [reply pkgs/development/python-modules/seasonal/default.nix]: All fixed!

**Comment by mbalatsko** [comment]
@natsukium all fixed!

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 256348` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.seasonal</li>
    <li>python310Packages.seasonal.dist</li>
    <li>python311Packages.seasonal</li>
    <li>python311Packages.seasonal.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/seasonal/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    owner = "welch";
+    repo = "seasonal";
+    # There are no tags or releases, but this commit corresponds to the 0.3.1 version
+    # PyPI project contains only a wheel
+    rev = "2a2396014d46283d0c7aff34cde5dafb6c462c58";
+    hash = "sha256-8YedGylH70pI0OyefiS1PG1yc+sg+tchlgcuNvxcNqE=";
+  };
+
+  patches = [
+    ./remove-pytest-runner-from-setup-py.patch
```
Comment (natsukium) [inline pkgs/development/python-modules/seasonal/default.nix:30]: Can we use `substituteInPlace` here?

Comment (mbalatsko) [reply pkgs/development/python-modules/seasonal/default.nix]: Done

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.seasonal python311Packages.seasonal

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**